src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,95 @@
+import React, { Children, Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import TabContent from './TabContent';
+import TabItem from './TabItem';
+import TabMenu from './TabMenu';
+
+class Tabs extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        const { activeTabId } = props;
+
+        this.state = {
+            activeTabId,
+            derivedProp: activeTabId // eslint-disable-line react/no-unused-state
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.activeTabId === state.derivedProp) {
+            return null;
+        }
+        const { activeTabId } = props;
+        return {
+            activeTabId,
+            derivedProp: activeTabId
+        };
+    }
+
+    handleTabChange(e, tabId) {
+        if (e && e.preventDefault) {
```

Comment:
Can't understand why do we need this check. In what scenario `e` can be not an Event?

Reply:
Not really necessary indeed, but I reckoned we might as well check to be 100% sure. If you think it's better not to check I'll remove it.

Reply:
Updated